script_editor/app/views/plays/show.html.erb

```diff
@@ -147,7 +159,87 @@
                         }
 
                     })
-                    //Change button text from Show to Hide or vice versa 
+                }
+
+                //formatPage handles adjusting the look of the script when switching between concise and full views
+                function formatPage(){
+                    script = document.querySelector(".script-main");
+                    var elements = script.querySelectorAll("[data-linenum]")
+                    var index = 0
+                    var newLineNum = 1
+                    while (index < elements.length)
+                    {
+                        var doCut = true
+                        var firstLineElm = index
+                        //Iterates through each line until a line break
+                        while(elements[index].className != "newLine")
+                        {
+                            ++index
+                            if (elements[index].dataset.cut == "false")
+                            {
+                                doCut = false // we don't want to cut the linebreak at the end of the line
+                            }
+                        }
+
+                        //Decides what to do once you're at a lineBreak
+                        if (doCut && toggleBtn.curState == "showing") //User wishes to hide edits 
+                        {
+                            elements[index].style.display = 'none'
+                        }
+                        else
+                        {
+                            elements[index].style.display = 'initial'
+
+                            //Create a new line number to display 
+                            var newNode = document.createElement("p")
+                            newNode.className = 'lineNum'
+                            newNode.id = "newLineNum"
+
+                            var newContent = document.createTextNode(newLineNum.toString())
+                            newNode.appendChild(newContent)
+                            if (newLineNum % 5 != 0) //we only wish to display every 5 line numbers 
```

Comment:
This is good and overall clear, but if we only want to display every 5 lines, we can wrap the entire new line num section in this if statement (except for the `++newLineNum`) and not waste time making an extra div we don't want to display. In other words, why not only make every fifth line?